libmicrohttpd does not compile with latest version of Xcode on macos26

At https://github.com/kiwix/kiwix-build/actions/runs/26363854555/job/77611883167

```
[5/15] /Applications/Xcode_26.5.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Ilibmicrohttpd.a.p -I. -I../../SOURCE/libmicrohttpd-0.9.76 -I../../SOURCE/libmicrohttpd-0.9.76/src/include -I/Users/runner/BUILD_x86-apple-ios-simulator/INSTALL/include -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -O3 -D_GNU_SOURCE -isysroot /Applications/Xcode_26.5.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator26.5.sdk -arch x86_64 -target x86-apple-ios-simulator -miphoneos-version-min=15.0 -MD -MQ libmicrohttpd.a.p/src_microhttpd_daemon.c.o -MF libmicrohttpd.a.p/src_microhttpd_daemon.c.o.d -o libmicrohttpd.a.p/src_microhttpd_daemon.c.o -c ../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/daemon.c
FAILED: [code=1] libmicrohttpd.a.p/src_microhttpd_daemon.c.o 
/Applications/Xcode_26.5.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Ilibmicrohttpd.a.p -I. -I../../SOURCE/libmicrohttpd-0.9.76 -I../../SOURCE/libmicrohttpd-0.9.76/src/include -I/Users/runner/BUILD_x86-apple-ios-simulator/INSTALL/include -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -O3 -D_GNU_SOURCE -isysroot /Applications/Xcode_26.5.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator26.5.sdk -arch x86_64 -target x86-apple-ios-simulator -miphoneos-version-min=15.0 -MD -MQ libmicrohttpd.a.p/src_microhttpd_daemon.c.o -MF libmicrohttpd.a.p/src_microhttpd_daemon.c.o.d -o libmicrohttpd.a.p/src_microhttpd_daemon.c.o -c ../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/daemon.c
../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/daemon.c:6610:11: error: call to undeclared function 'pipe2'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 6610 |     if (! MHD_itc_init_ (daemon->itc))
      |           ^
../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/mhd_itc.h:163:33: note: expanded from macro 'MHD_itc_init_'
  163 | #  define MHD_itc_init_(itc) (! pipe2 ((itc).fd, O_CLOEXEC | O_NONBLOCK))
      |                                 ^
../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/daemon.c:6610:11: note: did you mean 'pipe'?
../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/mhd_itc.h:163:33: note: expanded from macro 'MHD_itc_init_'
  163 | #  define MHD_itc_init_(itc) (! pipe2 ((itc).fd, O_CLOEXEC | O_NONBLOCK))
      |                                 ^
/Applications/Xcode_26.5.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator26.5.sdk/usr/include/unistd.h:482:6: note: 'pipe' declared here
  482 | int      pipe(int [2]);
      |          ^
../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/daemon.c:7248:17: error: call to undeclared function 'pipe2'; ISO C99 and later do not support implicit function declarations [-Wimplicit-function-declaration]
 7248 |           if (! MHD_itc_init_ (d->itc))
      |                 ^
../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/mhd_itc.h:163:33: note: expanded from macro 'MHD_itc_init_'
  163 | #  define MHD_itc_init_(itc) (! pipe2 ((itc).fd, O_CLOEXEC | O_NONBLOCK))
      |                                 ^
2 errors generated.
[6/15] /Applications/Xcode_26.5.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Ilibmicrohttpd.a.p -I. -I../../SOURCE/libmicrohttpd-0.9.76 -I../../SOURCE/libmicrohttpd-0.9.76/src/include -I/Users/runner/BUILD_x86-apple-ios-simulator/INSTALL/include -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -O3 -D_GNU_SOURCE -isysroot /Applications/Xcode_26.5.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator26.5.sdk -arch x86_64 -target x86-apple-ios-simulator -miphoneos-version-min=15.0 -MD -MQ libmicrohttpd.a.p/src_microhttpd_mhd_mono_clock.c.o -MF libmicrohttpd.a.p/src_microhttpd_mhd_mono_clock.c.o.d -o libmicrohttpd.a.p/src_microhttpd_mhd_mono_clock.c.o -c ../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/mhd_mono_clock.c
[7/15] /Applications/Xcode_26.5.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Ilibmicrohttpd.a.p -I. -I../../SOURCE/libmicrohttpd-0.9.76 -I../../SOURCE/libmicrohttpd-0.9.76/src/include -I/Users/runner/BUILD_x86-apple-ios-simulator/INSTALL/include -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -O3 -D_GNU_SOURCE -isysroot /Applications/Xcode_26.5.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator26.5.sdk -arch x86_64 -target x86-apple-ios-simulator -miphoneos-version-min=15.0 -MD -MQ libmicrohttpd.a.p/src_microhttpd_mhd_sockets.c.o -MF libmicrohttpd.a.p/src_microhttpd_mhd_sockets.c.o.d -o libmicrohttpd.a.p/src_microhttpd_mhd_sockets.c.o -c ../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/mhd_sockets.c
[8/15] /Applications/Xcode_26.5.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Ilibmicrohttpd.a.p -I. -I../../SOURCE/libmicrohttpd-0.9.76 -I../../SOURCE/libmicrohttpd-0.9.76/src/include -I/Users/runner/BUILD_x86-apple-ios-simulator/INSTALL/include -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -O3 -D_GNU_SOURCE -isysroot /Applications/Xcode_26.5.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator26.5.sdk -arch x86_64 -target x86-apple-ios-simulator -miphoneos-version-min=15.0 -MD -MQ libmicrohttpd.a.p/src_microhttpd_mhd_send.c.o -MF libmicrohttpd.a.p/src_microhttpd_mhd_send.c.o.d -o libmicrohttpd.a.p/src_microhttpd_mhd_send.c.o -c ../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/mhd_send.c
[9/15] /Applications/Xcode_26.5.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -Ilibmicrohttpd.a.p -I. -I../../SOURCE/libmicrohttpd-0.9.76 -I../../SOURCE/libmicrohttpd-0.9.76/src/include -I/Users/runner/BUILD_x86-apple-ios-simulator/INSTALL/include -fvisibility=hidden -fdiagnostics-color=always -Wall -Winvalid-pch -O3 -D_GNU_SOURCE -isysroot /Applications/Xcode_26.5.app/Contents/Developer/Platforms/iPhoneSimulator.platform/Developer/SDKs/iPhoneSimulator26.5.sdk -arch x86_64 -target x86-apple-ios-simulator -miphoneos-version-min=15.0 -MD -MQ libmicrohttpd.a.p/src_microhttpd_connection.c.o -MF libmicrohttpd.a.p/src_microhttpd_connection.c.o.d -o libmicrohttpd.a.p/src_microhttpd_connection.c.o -c ../../SOURCE/libmicrohttpd-0.9.76/src/microhttpd/connection.c
ninja: build stopped: subcommand failed.


Traceback (most recent call last):
  File "/Users/runner/work/kiwix-build/kiwix-build/.github/scripts/build_release_nightly.py", line 35, in <module>
    run_kiwix_build(target, config=COMPILE_CONFIG, make_release=MAKE_RELEASE)
    ~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/runner/work/kiwix-build/kiwix-build/.github/scripts/common.py", line 239, in run_kiwix_build
    subprocess.check_call(command, cwd=str(HOME), env=env)
    ~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Library/Frameworks/Python.framework/Versions/3.13/lib/python3.13/subprocess.py", line 419, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['kiwix-build', 'libkiwix', '--hide-progress', '--fast-clone', '--assume-packages-installed', '--use-target-arch-name', '--config', 'apple_all_static', '--make-release']' returned non-zero exit status 1.
Error: Process completed with exit code 1.

```

@veloman-yunkan Not clear what happens here, but I guess this version of the compiler is just stricter than the older one... and because we have a version of libmicrohttpd which is already 4 years old... Could you please confirm the analysis and let know what can be done?

--- a/kiwixbuild/dependencies/libmicrohttpd.py
+++ b/kiwixbuild/dependencies/libmicrohttpd.py
@@ -22,6 +22,7 @@ class MicroHttpd(Dependency):
             "libmicrohttpd_meson_pkgconfig.patch",
             "libmicrohttpd_meson_timeval_tvsec_size.patch",
             "libmicrohttpd_meson_winet6.patch",
+            "libmicrohttpd_meson_pipe2.patch",
         ]
 
     class Builder(MesonBuilder):

--- a/kiwixbuild/patches/libmicrohttpd_meson_pipe2.patch
+++ b/kiwixbuild/patches/libmicrohttpd_meson_pipe2.patch
@@ -1,0 +1,14 @@
+--- libmicrohttpd-0.9.76/meson.build.orig       2026-05-27 13:37:50
++++ libmicrohttpd-0.9.76/meson.build    2026-05-27 13:44:16
+@@ -152,7 +152,10 @@
+ )
+ cdata.set('SIZEOF_UINT64_T', 8)
+
+-cdata.set('HAVE_PIPE2_FUNC', cc.has_function('pipe2'))
++if host_machine.system() != 'darwin'
++  cdata.set('HAVE_PIPE2_FUNC', cc.has_function('pipe2'))
++endif
++
+ cdata.set('MHD_HAVE___BUILTIN_BSWAP32', cc.has_function('__builtin_bswap32'))
+ cdata.set('MHD_HAVE___BUILTIN_BSWAP64', cc.has_function('__builtin_bswap64'))
+

--- a/kiwixbuild/versions.py
+++ b/kiwixbuild/versions.py
@@ -33,7 +33,7 @@ release_versions = {
 
 # This is the "version" of the whole base_deps_versions dict.
 # Change this when you change base_deps_versions.
-base_deps_meta_version = "22"
+base_deps_meta_version = "23"
 
 base_deps_versions = {
     "zlib": "1.3.1",

